### DISCUSSION TITLE
patterns/strategy/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#69] [Announcements]
# patterns/strategy/

Also known as Policy Intent Define a family of algorithms, encapsulate each one, and make them interchangeable. Strategy lets the algorithm vary independently of the clients that use it. Explanation Real-world example Slaying dragons is a dangerous job. With experience, it becomes easier. Veteran dragonslayers have developed different fighting strategies against different types of dragons.

https://java-design-patterns.com/patterns/strategy/

**Comment by shandianlala** [comment]
it is great
